### DISCUSSION TITLE
Problem with IVirtualPageController and LastChanceContentFinder order

### DISCUSSION THREAD

**Discussion by Andrei-** [umbraco/Umbraco-CMS#12412] [Features and ideas]
I'm trying to make a virtual products node, like the example in the documentation here:
[https://our.umbraco.com/Documentation/Reference/Routing/custom-routes](https://our.umbraco.com/Documentation/Reference/Routing/custom-routes)

My problem is - how do I serve the 404 content, normally served from LastChanceContentFinder when my product does not exist? 
Since LastChanceContentFinder executes prior to my IVirtualPageController and in the controller it's too late.
Normally I would like to validate if my product exists in a content finder,  and if necessary, prevent IVirtualPageController execution and serve the NotFound content node.

Validating a specific route in a content finder would be ok-ish, but if you have a ton of matching routes to validate, it's no longer ideal.

**Comment by nul800sebastiaan** [comment]
Hi there @Andrei- - this sounds like an excellent question for the forums where our friendly community can help you find the best solution for your requirements. 

Make sure to head on over to https://our.umbraco.com and ask follow up questions there! 👍
